### PR TITLE
Add javadoc

### LINKED ISSUE #5: Add javadoc to public API

### PR DESCRIPTION
**Pull request by FaustXVI**
Closes #5 

- [x] Docker annotation
- [x] Port annotation
- [x] Environment annotation
- [x] WaitFor annotation
- [x] DockerExtension class

### CODE REVIEW

File: src/main/java/com/github/junit5docker/DockerExtension.java [hunk 1]
```diff
@@ -7,7 +7,7 @@
 import java.util.HashMap;
 import java.util.concurrent.*;
 
-public class DockerExtension implements BeforeAllCallback, AfterAllCallback {
+class DockerExtension implements BeforeAllCallback, AfterAllCallback {
```
Comment (FaustXVI) [inline src/main/java/com/github/junit5docker/DockerExtension.java:10]: The DockerExtension does not have to be public so we can avoid useless documentation.

File: src/main/java/com/github/junit5docker/Environment.java [hunk 2]
```diff
@@ -1,7 +1,19 @@
 package com.github.junit5docker;
 
+/**
+ * Describes an environment variable to pass to the docker's container.
```
Comment (vdemeester) [inline src/main/java/com/github/junit5docker/Environment.java]: > Describes an environment variable to set for a docker conatiner

File: src/main/java/com/github/junit5docker/Port.java [hunk 3]
```diff
@@ -1,7 +1,21 @@
 package com.github.junit5docker;
 
+/**
+ * Describes a port binding to send to docker.
```
Comment (vdemeester) [inline src/main/java/com/github/junit5docker/Port.java]: It's a little bit different than the one on `Environment`.

> Describes a port binding for a docker container.

File: src/main/java/com/github/junit5docker/Port.java [hunk 4]
```diff
@@ -1,7 +1,21 @@
...
+ *
+ * @since 1.0
+ */
 public @interface Port {
+
+    /**
+     * @return the port number exposed on localhost.
```
Comment (vdemeester) [inline src/main/java/com/github/junit5docker/Port.java]: s/localhost/the host/

File: src/main/java/com/github/junit5docker/Port.java [hunk 5]
```diff
@@ -1,7 +1,21 @@
...
+     * This is the port number to use in the test code in order to send data to the container.
```
Comment (vdemeester) [inline src/main/java/com/github/junit5docker/Port.java]: It's not only for sending data :angel: (to `exchange` or to `connect` or something like that).

File: src/main/java/com/github/junit5docker/WaitFor.java [hunk 6]
```diff
@@ -1,10 +1,26 @@
...
+ * <p>Asks JUnit-docker to wait for something to appear in the container's log before running any test.</p>
+ *
+ * @since 1.0
+ */
 public @interface WaitFor {
 
-    public static String NOTHING = "";
+    /**
+     * <p>Special value.
+     * If used, then JUnit-docker will not wait for anything and just start the tests.</p>
```
Comment (vdemeester) [inline src/main/java/com/github/junit5docker/WaitFor.java]: If used, JUnit-docker does not…

**Review by vdemeester — CHANGES_REQUESTED** [review]
Few comments :anchor:

File: src/main/java/com/github/junit5docker/Port.java (hunk 4 above)
Comment (FaustXVI) [reply src/main/java/com/github/junit5docker/Port.java]: Well, actually in our case, the host is localhost. And I'd like to make it explicit

Comment (vdemeester) [reply src/main/java/com/github/junit5docker/Port.java]: @FaustXVI hum, this is the tricky part, you don't know if it's localhost or anything really. It depends on the `docker` daemon you're talking to.

Comment (FaustXVI) [reply src/main/java/com/github/junit5docker/Port.java]: Right ! It the socket that is local (i.e on localhost) but the real host can be distant !
My bad :)

**Review by vdemeester — APPROVED** [review]
LGTM 🐸
Probably should squash a few commits 👼

**Pull request merged**